rundown/views.py

```diff
@@ -0,0 +1,86 @@
+from rest_framework.views import APIView
+from django.shortcuts import get_object_or_404
+from event.models import Event
+from .models import Rundown
+from .serializers import RundownSerializer
+from rest_framework.response import Response
+from .validators import validate_rundown_data, is_valid_updated_data
+from drf_yasg.utils import swagger_auto_schema
+from drf_yasg import openapi
+
+class RundownCreateView(APIView):
+
+    @swagger_auto_schema(
+        request_body=openapi.Schema(
+            type=openapi.TYPE_OBJECT,
+            required=['event_id', 'rundown_data'],
+            properties={
+                'event_id': openapi.Schema(type=openapi.TYPE_INTEGER, description='Event ID'),
+                'rundown_data': openapi.Schema(
+                    type=openapi.TYPE_ARRAY,
+                    description='List of rundowns',
+                    items=openapi.Items(
+                        type=openapi.TYPE_OBJECT,
+                        properties={
+                            'description': openapi.Schema(type=openapi.TYPE_STRING, description='Activity that time in rundown'),
+                            'start_time': openapi.Schema(type=openapi.TYPE_STRING, description='Start time of the activity'),
+                            'end_time': openapi.Schema(type=openapi.TYPE_STRING, description='End time of the activity'),
+                        }
+                    )
+                ),
+            },
+        ),
+        responses={201: RundownSerializer(many=True), 400: 'Bad Request', 500: 'Internal Server Error'}
+    )
+    def post(self, request):
+        event_id = request.data.get('event_id')
+        rundown_data = request.data.get('rundown_data')
+
+        event = get_object_or_404(Event, id=event_id)
+
+        if Rundown.objects.filter(event=event).exists():
+            return Response({"error": "Rundown for that event already exists"}, status=400)
+
+        if not validate_rundown_data(rundown_data):
+            return Response({"error": "Invalid rundown data"}, status=400)
+        
+        order = 1
+
+        for data in rundown_data:
+            data["event_id"] = event_id
+            data["rundown_order"] = order
+            order += 1
+        
+        created_rundown = Rundown.objects.bulk_create([Rundown(**data) for data in rundown_data])
+        rundown_serializers = RundownSerializer(created_rundown, many=True)
+        return Response(rundown_serializers.data, status=201)    
+    
+class RundownUpdateView(APIView):
```

Comment:
I think this is should be RundownDetailView instead, so that others, in here Swas could just add method delete to delete the rundown. But overall great job.

Reply:
hmmm ok